### PR TITLE
Feat: generate docs for reference automatically

### PR DESCRIPTION
**Pull request by wonderflow**
Signed-off-by: Jianbo Sun <jianbo.sjb@alibaba-inc.com>

### Description of your changes

1. enhance `vela def doc-gen`, support for:
   - Generate documentation for ComponentDefinition webservice
    ```
    vela def doc-gen webservice -n vela-system
    ```
   - Generate documentation for local CUE Definition file webservice.cue
   ```
    vela def doc-gen webservice.cue
   ```

2. add scripts for generate vela built-in reference docs, add some rules for generated docs

https://github.com/kubevela/kubevela.io/pull/857/files#diff-83f8c791b2a42df6f97d94ab7ca7559d6ede1a6dbaa6bc27b2acc2e23e6a37cc

3. Refactor the code, align with `vela show`

4. trim prefix of definition annoatations/label if it has oam.dev suffix, affected labels:
   - "custom.definition.oam.dev/alias.config.oam.dev" => "alias.config.oam.dev"
   - "custom.definition.oam.dev/catalog.config.oam.dev" => "catalog.config.oam.dev"
   - "custom.definition.oam.dev/multi-cluster.config.oam.dev" => "multi-cluster.config.oam.dev"
   - "custom.definition.oam.dev/type.config.oam.dev" => "type.config.oam.dev"

5. A definition annoation ("definition.oam.dev/example-url") was added, it can be used to describe the definition example file.
    - "definition.oam.dev/example-url": "https://kubevela.io/examples/applications/create-namespace.yaml"

Now, a definition can describe itself, for example:

```
"testdef": {
	type: "component"
	annotations: {
		"definition.oam.dev/example-url": "https://kubevela.io/examples/applications/create-namespace.yaml"
	}
	labels: {}
	description: "K8s-objects allow users to specify raw K8s objects in properties"
	attributes: workload: type: "autodetects.core.oam.dev"
}
template: {
	output: parameter.objects[0]

	outputs: {
		for i, v in parameter.objects {
			if i > 0 {
				"objects-\(i)": v
			}
		}
	}
	parameter: {
		// +usage=A test key
		objects: [...{}]
	}
}
```

It contains the description as field, examples loaded from that url and parameter specification generated  fromtemplate.paramer.

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/kubevela/kubevela/blob/master/contribute/create-pull-request.md).
- [x] [Related Docs](https://github.com/kubevela/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

File: references/plugins/def-doc/component/alibaba-ack.eg.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+```yaml
```
Comment (barnettZQG) [inline references/plugins/def-doc/component/alibaba-ack.eg.md:1]: Here markdown documents need manual maintenance?

Comment (wonderflow) [reply references/plugins/def-doc/component/alibaba-ack.eg.md:1]: yes, examples need to be provided manully

**Review by barnettZQG — APPROVED** [review]

File: pkg/definition/definition.go [hunk 2]
```diff
@@ -284,15 +284,23 @@ func (def *Definition) FromCUE(val *cue.Value, templateString string) error {
 					return err
 				}
 				for _k, _v := range _annotations {
-					annotations[UserPrefix+_k] = _v
+					if strings.Contains(_k, "oam.dev") {
```
Comment (Somefive) [inline pkg/definition/definition.go:287]: So it means we allow user to edit system generated annotations? Like addon.oam.dev?

File: pkg/utils/load.go [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+	j "encoding/json"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/oam-dev/kubevela/pkg/utils/common"
+)
+
+// ReadRemoteOrLocalPath will read a path remote or locally
+func ReadRemoteOrLocalPath(pathOrURL string, saveLocal bool) ([]byte, error) {
```
Comment (Somefive) [inline pkg/utils/load.go:30]: pkg/utils/file.go LoadDataFromPath looks similar to this function?

File: references/cli/up.go [hunk 4]
```diff
@@ -39,6 +39,7 @@ import (
 	"github.com/oam-dev/kubevela/pkg/controller/core.oam.dev/v1alpha2/application"
 	"github.com/oam-dev/kubevela/pkg/controller/utils"
 	"github.com/oam-dev/kubevela/pkg/oam"
+	utils2 "github.com/oam-dev/kubevela/pkg/utils"
```
Comment (Somefive) [inline references/cli/up.go:42]: velautils? Or pkgutils?

File: references/plugins/i18n.go [hunk 5]
```diff
@@ -16,120 +16,172 @@ limitations under the License.
...
+const (
+	// LangEn is english, the default language
+	LangEn Language = "English"
+	// LangZh is Chinese
+	LangZh Language = "Chinese"
+)
+
+// I18n will automatically get translated data
+type I18n struct {
+	lan Language
```
Comment (Somefive) [inline references/plugins/i18n.go:46]: lang?

File: pkg/definition/definition.go (hunk 2 above)
Comment (wonderflow) [reply pkg/definition/definition.go:287]: it's not about edit, it's generated without redundant prefix

File: pkg/utils/load.go (hunk 3 above)
Comment (wonderflow) [reply pkg/utils/load.go:30]: they're not exactly the same, not related with this PR

File: references/cli/up.go (hunk 4 above)
Comment (wonderflow) [reply references/cli/up.go:42]: good idea

File: references/plugins/i18n.go (hunk 5 above)
Comment (wonderflow) [reply references/plugins/i18n.go:46]: good catch

**Review by Somefive — APPROVED** [review]

**Pull request merged**